src/CounterInterceptor/ClientCounter.cs

```diff
@@ -0,0 +1,89 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using System.Numerics;
+
+public class ClientCounter
+{
+    private const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    private const string NumberOfSignals = "numOfSignals";
+    private const string NumberOfQueries = "numOfQueries";
+    private static Dictionary<string, Dictionary<string, BigInteger?>> perWorkflowIdDictionary =
+        new Dictionary<string, Dictionary<string, BigInteger?>>();
+
+    public static string Info()
+    {
+        string result = string.Empty;
+        foreach (var item in perWorkflowIdDictionary)
+        {
+            var info = item.Value;
+            result = result +
+                "\n** Workflow ID: " + item.Key +
+                "\n\tTotal Number of Workflow Exec: " + info[NumberOfWorkflowExecutions] +
+                "\n\tTotal Number of Signals: " + info[NumberOfSignals] +
+                "\n\tTotal Number of Queries: " + info[NumberOfQueries];
+        }
+
+        return result;
+    }
+
+    public static BigInteger? NumOfWorkflowExecutions(string workflowId)
```

Comment:
Assuming the field is properly changed to `private static Dictionary<string, WorkflowCounts> workflowCounts`, a simple `public static WorkflowCounts? GetWorkflowCounts(string workflowId) => workflowCounts.TryGetValue(workflowId, out var counts) ? counts : null` will work